backend/project-service/src/main/kotlin/project/services/QueryService.kt

```diff
@@ -23,6 +25,32 @@ data class ProjectForVerification(
 class QueryService(
     private val projects: ProjectService
 ) {
+    suspend fun searchProjectPaths(ctx: DBContext, actor: Actor, query: String): Map<String, String> {
+        val role = if (actor is Actor.User) actor.principal.role else Role.USER
+        if (role !in Roles.ADMIN) {
+            throw RPCException.fromStatusCode(HttpStatusCode.Forbidden, "User not admin")
+        }
+        return ctx.withSession { session ->
+            val results = session.sendPreparedStatement(
+                {
+                    setParameter("query", query)
+                },
+                """
+                    SELECT *
+                    FROM projects
+                    WHERE title LIKE '%' || :query || '%'
+                """
+            ).rows
+
+            val paths = mutableMapOf<String, String>()
+            results.forEach {
+                val path = viewAncestors(ctx, actor, it.getField(ProjectTable.id)).map { it.title }
+                paths[it.getField(ProjectTable.id)] = path.joinToString("/")
+            }
+            paths
```

Comment:
Comment above likely indicates that this needs to be changed but the code could be rewritten as:

```suggestion
            results.map {
                val path = viewAncestors(ctx, actor, it.getField(ProjectTable.id)).map { it.title }
                it.getField(ProjectTable.id) to path.joinToString("/")
            }.toMap()
```